osf/models/node.py

```diff
@@ -1486,6 +1486,12 @@ def has_pointers_recursive(self):
                 return True
         return False
 
+    def add_affiliations(self, user, new):
+        # add affiliations from the original to the forked or templated node
+        user_affils = user.affiliated_institutions.values_list('id', flat=True)
+        for affiliation in self.affiliated_institutions.filter(id__in=user_affils):
+            new.affiliated_institutions.add(affiliation)
```

Comment:
So here we loop through all the `affiliated_institutions` on the original node and add to the new node provided the user performing the operation belongs to that institution.

So to clarify, this is what we expect in the example below? Institution C is _not_ copied to the new node because it isn't on the original.  Institution B is _not_ copied to the new node because User doesn't have that affiliation.  
Original -> A, B
User - > A, C
New -> A

Reply:
Yes I do believe that is the expected behavior, thank you for the concise example! I will double check with Sara.

Reply:
Good catch, checked with Sara, and it should indeed be:

Original -> A, B
User - > A, C
New -> A, C

Updating!